server/src/middleware/merchant-auth.ts

```diff
@@ -29,13 +38,23 @@ const merchantAuth = async (
       throw new Error('Missing Authorization header.');
 
     const [_, firebaseIdToken] = req.headers.authorization.split(' ');
-    const decodedFirebaseIdToken = await admin
-      .auth()
-      .verifyIdToken(firebaseIdToken);
-    req.decoded = decodedFirebaseIdToken;
+    const verifiedFirebaseUid = await getVerifiedUid(firebaseIdToken);
+
+    let firebaseUid;
+    if (req.body.firebaseUid !== undefined) {
+      firebaseUid = req.body.firebaseUid;
+    } else if (req.body.merchantId !== undefined) {
+      const merchant = await merchantService.getMerchant(req.body.merchantId);
+      firebaseUid = merchant.firebaseUid;
+    }
+
+    if (firebaseUid !== undefined && firebaseUid !== verifiedFirebaseUid)
```

Comment:
could firebaseUid be undefined?

Reply:
@ifhuang Yep, that's if neither of the 2 if-elseif conditions above are satisfied. It's related to @xlai20's comment above.